calico/openstack/mech_calico.py

```diff
@@ -196,14 +206,148 @@ def _init_state(self):
 
         self._my_pid = current_pid
 
-        # Start our resynchronization process.  Just in case we ever get two
-        # threads running, use an epoch counter to tell the old thread to die.
+        # Start our resynchronization process and status updating. Just in
+        # case we ever get two same threads running, use an epoch counter to
+        # tell the old thread to die.
         # This is defensive: our greenlets don't actually seem to get forked
         # with the process.
         # We deliberately do this last, to ensure that all of the setup above
         # is complete before we start running.
         self._epoch += 1
         eventlet.spawn(self.periodic_resync_thread, self._epoch)
+        eventlet.spawn(self._status_updating_thread, self._epoch)
+
+    def _status_updating_thread(self, expected_epoch):
+        """
+        This method acts as a status updates handler logic for the
+        Calico mechanism driver. Watches for felix updates in etcd
+        and passes info to Neutron database.
+        """
+        LOG.info("Handle status updating thread started")
+        self._db_context = ctx.get_admin_context()
+
+        # We only read initial felix instances in etcd if we are master.
+        if self.transport.is_master:
+            self._client = etcd.Client(host=cfg.CONF.calico.etcd_host,
+                                       port=cfg.CONF.calico.etcd_port)
+            if self._epoch == expected_epoch:
+                self._register_initial_felixes()
+
+        while self._epoch == expected_epoch:
+            # Only handle updates if we are the master node.
+            if self.transport.is_master:
+                LOG.info("I am master: polling felix updates in etcd")
+                self._poll_and_handle_felix_updates(expected_epoch)
+            else:
+                # Short sleep interval before we check if we've become
+                # the master.
+                eventlet.sleep(MASTER_CHECK_INTERVAL_SECS)
+        else:
+            LOG.warning("Unexpected fork. "
```

Comment:
`fork` has s technical meaning so best to avoid it.  I'd probably say: "Unexpected: epoch changed; update thread exiting"